numba/inline_closurecall.py

```diff
@@ -958,16 +958,32 @@ def inline_array(array_var, expr, stmts, list_vars, dels):
         stmts.append(_new_definition(func_ir, size_tuple_var,
                  ir.Expr.build_tuple(items=[size_var], loc=loc), loc))
 
+        nptype = types.DType(dtype)
+
         empty_func = ir.Var(scope, mk_unique_var("empty_func"), loc)
         fnty = get_np_ufunc_typ(np.empty)
-        sig = context.resolve_function_type(fnty, (size_typ,), {})
-        typemap[empty_func.name] = fnty #
+        sig = context.resolve_function_type(fnty, (size_typ,), {'dtype':nptype})
+
+        typemap[empty_func.name] = fnty
 
         stmts.append(_new_definition(func_ir, empty_func,
                          ir.Global('empty', np.empty, loc=loc), loc))
 
-        empty_call = ir.Expr.call(empty_func, [size_var], {}, loc=loc)
-        calltypes[empty_call] = typing.signature(array_typ, size_typ)
+        g_np_var = ir.Var(scope, mk_unique_var("$np_g_var"), loc)
```

Comment:
Perhaps a comment or two as to what this is doing would be good ? Purely for ease should this be revisited.

Reply:
Done.